kit/src/main/java/com/oracle/javafx/scenebuilder/kit/skeleton/SkeletonFileNameProposal.java

```diff
@@ -0,0 +1,192 @@
+         * well. This strongly depends on how users work with SceneBuilder.
+         * 
+         */
+        List<Path> sourcePackages = List.of(Paths.get("src/main"));
+        for (Path sourcePackage : sourcePackages) {
+
+            String resources = sourcePackage.resolve("resources").toString();
+            String java      = sourcePackage.resolve("java").toString();
+            String kotlin    = sourcePackage.resolve("kotlin").toString();
+
+            if (location.contains(resources)) {
+                switch (language) {
+                case JAVA:
+                    location = location.replace(resources, java);
+                    break;
+                case KOTLIN:
+                    location = location.replace(resources, kotlin);
+                    break;
+                }
+            }
+        }
+
+        return new File(location);
+    }
+
+    private File createFileFromControllerName(URL fxmlLocation, String fxControllerName) {
+
+        String directory = obtainUserDirectory();
+        if (null != fxmlLocation) {
+            URI uri = resolveURI(fxmlLocation);
+            Path location = Paths.get(uri).toAbsolutePath().getParent();
+            if (Files.exists(location)) {
+                directory = location.toString();
+            }
+        }
+
+        String simpleControllerClassName = extractSimpleControllerName(fxControllerName);
+        String controllerFileName = simpleControllerClassName + language.getExtension();
+        return new File(directory, controllerFileName);
+    }
+
+    private URI resolveURI(URL fxmlLocation) {
+        try {
+            return fxmlLocation.toURI();
+        } catch (URISyntaxException e) {
+            File usersDir = new File(obtainUserDirectory()).getAbsoluteFile();
+            return usersDir.toURI();
+        }
+    }
+
+    private File createFileAccordingToFxml(URL fxmlLocation) {
+
+        URI  uri = resolveURI(fxmlLocation);
+        Path fxmlFile = Paths.get(uri);        
+        String fxmlResource = fxmlFile.toString();
+        String controllerSuffix = "Controller" + language.getExtension();
+
+        int lastDot = fxmlResource.lastIndexOf('.');
+        if (lastDot > -1) {
+            String newFileName = fxmlResource.substring(0, lastDot) + controllerSuffix;
+            return Paths.get(newFileName).toFile();
+        }
+
+        return Paths.get(fxmlResource + controllerSuffix).toFile();
+
+    }
+
+    private File createFileInUsersDir() {
+        String fileName = DEFAULT_CONTROLLER_CLASS_NAME + language.getExtension();
+        String usersDirectory = obtainUserDirectory();
+        return buildFileName(usersDirectory, fileName);
+    }
+
+    private String obtainUserDirectory() {
+        String usersDirectory = System.getProperty("user.home");
+        if (usersDirectory == null || usersDirectory.isBlank()) {
+            usersDirectory = "./";
```

Comment:
This is most probably going to fail on Windows. BTW, can `user.home` be null?

Reply:
Well, here I am not sure if this can happen. If it cannot be null or empty, there is no need for checks. So I removed those.